.golangci.yml

```diff
@@ -0,0 +1,59 @@
+# This files contains all configuration options for analysis running.
+# More details please refer to: https://golangci-lint.run/usage/configuration/
+
+run:
+  # timeout for analysis, e.g. 30s, 5m, default is 1m
+  timeout: 5m
+
+  # which dirs to skip: issues from them won't be reported;
+  # can use regexp here: generated.*, regexp is applied on full path;
+  # default value is empty list, but default dirs are skipped independently
+  # from this option's value (see skip-dirs-use-default).
+  # "/" will be replaced by current OS file path separator to properly work
+  # on Windows.
+  skip-dirs:
+    -
+
+  # default is true. Enables skipping of directories:
+  #   vendor$, third_party$, testdata$, examples$, Godeps$, builtin$
+  skip-dirs-use-default: true
+
+  # One of 'readonly' and 'vendor'.
+  #  - readonly: the go command is disallowed from the implicit automatic updating of go.mod described above.
+  #              Instead, it fails when any changes to go.mod are needed. This setting is most useful to check
+  #              that go.mod does not need updates, such as in a continuous integration and testing system.
+  #  - vendor: the go command assumes that the vendor directory holds the correct copies of dependencies and ignores
+  #            the dependency descriptions in go.mod.
+  modules-download-mode: readonly
+linters:
+  enable:
+    # linters maintained by golang.org
+    - gofmt
+    - goimports
+    - govet
+    # linters default enabled by golangci-lint .
+    - deadcode
+    - errcheck
+    - gosimple
+    - ineffassign
+    - staticcheck
+    - structcheck
+    - typecheck
+    - unused
+    - varcheck
+    # other linters supported by golangci-lint.
+    #- gocyclo
+    #- gosec
+    - whitespace
+    #- revive
```

Comment:
these commented out linters contain too many errors or warnings, so just comment out temporarily, can be fixed with new PRs